### PR TITLE
(maint) Fix autodetected versions in pa_matchbatch

### PR DESCRIPTION
**Pull request by caseywilliams**
- Fixes regex for overrides files with more than one instance of the
  string 'puppet'
- Adds detection of versions based on common version files

### CODE REVIEW

**Review by ScottGarman — APPROVED** [review]

File: pa_matchbatch.sh [hunk 1]
```diff
@@ -72,17 +72,58 @@ getComponentRevMap() {
...
+# Find the fixVersion for a project in this release
 #
+# - If an overrides file was supplied, use the version number from that file
+# - Otherwise look for common version files
+# - Otherwise try to find the most recent version update in the git log
 getFixVerFor() {
-	if [[ -f  ${OVERRIDE_PATH} ]]; then
-		override=$(grep ${1}  ${OVERRIDE_PATH} | cut -d: -f2)
-		[[ -n ${override} ]] && { echo ${override}; return; }
+    # If an overrides file was supplied, look at it for a version number first
```
Comment (speedofdark) [inline pa_matchbatch.sh]: NIT: indenting of the first two possibilities is inconsistent with the other options

Comment (caseywilliams) [reply pa_matchbatch.sh]: Ah, this file uses tabs. I'll add a modeline and update

File: pa_matchbatch.sh [hunk 2]
```diff
@@ -72,17 +72,58 @@ getComponentRevMap() {
...
+    componentName="$(basename ${PWD})"
+    if [[ $componentName = "puppet-resource_api" ]]; then
+        # The resource api has a './lib/puppet/resource_api/version.rb' which contains a `VERSION = <version>` line
+        versionFile="./lib/puppet/resource_api/version.rb"
+    else
+        # puppet has a './lib/puppet/version.rb' which contains a `PUPPETVERSION = <version>` line
+        # hiera has a './lib/hiera/version.rb' which contains a `VERSION = <version>` line
+        versionFile="./lib/${componentName}/version.rb"
+    fi
+	if [[ -f ${versionFile} ]]; then
```
Comment (speedofdark) [inline pa_matchbatch.sh:111]: NIT here too.

**Review by speedofdark — APPROVED** [review]
LGTM after minor indenting fixed :)

**Review by speedofdark — APPROVED** [review]

**Comment by caseywilliams** [comment]
Ok, I've added another commit that makes the whitespace consistent (there were other sections that used spaces, too)

**Pull request merged**